### PR TITLE
adding manifest property for  token url configs

### PR DESCRIPTION
**Pull request by uhunnyslack**
###  Summary
To support more connectors, 3p auth manifest config requires some more new SDK fields

This PR is to add two new fields that helps the connector team unblock creating some more connectors.

1. `use_pkce` : boolean -> this is true for providers that uses PKCE:

2. `token_url_config` : { `use_basic_authentication_scheme_scheme` : boolean} to support providers that use this.

More discussions can be found in our slack instance: 

### Requirements (place an `x` in each `[ ]`)

* [X] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/deno-slack-sdk/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by WilliamBergamin** [comment]
@uhunnyslack I was working on #194 that aims to consolidate the provider args options `type` and the `type` found in the `manifest` we decided to put this PR on hold waiting to see if your changes could conflict with mine

Do you know if there will be a new provider other changes coming in the near future and would it be worth it to create a typescript type dedicated to PKCE providers?

File: src/providers/oauth2/types.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@ export type OAuth2ProviderIdentitySchema = {
   };
 };
 
+export type tokenUrlConfigSchema = {
+  "use_basic_authentication_scheme"?: boolean;
```
Comment (filmaj) [inline src/providers/oauth2/types.ts:12]: Let's add a JSdoc here to explain what this is used for.

File: .slack/apps.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (filmaj) [inline .slack/apps.json:1]: Please remove this file.

**Review by filmaj — CHANGES_REQUESTED** [review]
Overall LGTM, left some comments on removing the apps.json file and adding more docs.

Let's coordinate in Slack re: releasing this.

**Pull request closed without merging**